File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    driver.get("https://www.training-support.net/selenium/target-practice")

    print("Title of Target Practice page: ", driver.title)

    third_header = driver.find_element(By.XPATH, "//h3[@id='third-header']").text
    print(third_header)

    fifth_header_color=driver.find_element(By.XPATH, "//h5[text()='Fifth header']")\
        .value_of_css_property("color")
    print(fifth_header_color)

    violet_classes = driver.find_element(By.CSS_SELECTOR, ".ui.violet.button")\
        .get_attribute("class")
    print(violet_classes)

    grey_text = driver.find_element(By.CSS_SELECTOR, ".ui.grey.button").text
    print(grey_text)

    driver.close()

File: Selenium/Activities/Activity4.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();

        driver.get("https://www.training-support.net/selenium/target-practice");
        System.out.println("Title of Target Practice page: " + driver.getTitle());
        String third_header=driver.findElement(By
                .xpath("//h3[@id='third-header']")).getText();
        System.out.println(third_header);
        String fifth_header_color=driver.findElement(By
                .xpath("//h5[text()='Fifth header']"))
                .getCssValue("color");
        System.out.println(fifth_header_color);
        String violet_classes = driver.findElement(By
                .cssSelector(".ui.violet.button"))
                .getAttribute("class");
        System.out.println(violet_classes);
        String grey_text = driver.findElement(By
                        .cssSelector(".ui.grey.button")).getText();
        System.out.println(grey_text);
        driver.close();




    }
}
